### ISSUE TITLE
React Tooltip - Accessibility markup cleanup

### ISSUE THREAD

**Issue by carmacleod** [carbon-design-system/carbon-components-react#2193]
There are some unnecessary attributes in the accessibility-related markup for the React Tooltip.

For an overview, please see the screen capture of the markup, below.

Please note:
- if the user provides visible text content for the bx--tooltip__label, then the button should use `aria-labelledby` to point to its id, and if the user doesn't have visible text, then they need to provide an aria-label for the button
- the component should have a sensible default aria-label, like "Info" or "Help" (not "tooltip"), and this default should be published in the doc for the aria-label prop (although users should be encouraged to provide their own)

Please delete:
- title - it is completely unnecessary
- aria-owns - this attribute is not used in the tooltip pattern
- role="img" and aria-label="tooltip" on the svg - these are not necessary because the button label overrides the image label (or keep role="img" and aria-label[ledby] on the svg and remove aria-label[ledby] from the button)
- alt="tooltip" on the svg - just delete this - alt is not a valid attribute on svg elements
- aria-labelledby on the tooltip div - this is not used in the tooltip pattern

Please consider:
- try to use a button element instead of `div role="button" tabindex="0"` ... and use onclick instead of handling space and enter keys
- feel free to test with [this little test case](https://carmacleod.github.io/playground/tooltip-test.html) before switching to button. It uses a real button with (mostly) bx styles, and it looks ok.

![image](https://user-images.githubusercontent.com/3331913/55920628-8fc23600-5bc7-11e9-8de1-91b239d50323.png)

**Comment by paschalidi** [comment]
@carmacleod will look into this tomorrow!

**Comment by paschalidi** [comment]
>try to use a button element instead of div role="button" tabindex="0" ... and use onclick instead of handling space and enter keys

Is this part of the requirements 100 percent? I am happy to check this out then! Le me know!

**Comment by paschalidi** [comment]
@carmacleod you did an amazing job explaining what needs to be changed here!! Thanks a lot for your time and effort you reeeeally helped me out realise what needs to be done here! Amazing job, thank you very much!

**Comment by paschalidi** [comment]
@carmacleod can we please have a look what is left here?

I think in my understanding, only
>try to use a button element instead of div role="button" tabindex="0" ... and use onclick instead of handling space and enter keys

I am not sure if the above is a necessity of a "good to have" feature tho. 

Can you please when you get some time confirm?
Then I am happy to assign this to me and take care of it.

**Comment by dakahn** [comment]
@carmacleod 👋

**Comment by carmacleod** [comment]
@paschalidi Oh, sorry - I thought I answered this earlier! No, it's not necessary to switch to button. :)
(as long as all of the focus/keyboard/visual behavior for a button are provided by the div and its markup/js/css, which I believe they are in this case. :)

**Comment by paschalidi** [comment]
Hello to you both! Then I have the feeling that this issue can be closed. @dakahn @carmacleod can you please confirm?

**Comment by carmacleod** [comment]
Yep! Closing. Thanks! :tada:
